src/main/java/hudson/tasks/junit/SuiteResult.java

```diff
@@ -123,10 +124,19 @@
         SAXReader saxReader = new SAXReader();
         ParserConfigurator.applyConfiguration(saxReader,new SuiteResultParserConfigurationContext(xmlReport));
 
-        Document result = saxReader.read(xmlReport);
-        Element root = result.getRootElement();
+        FileInputStream xmlReportStream = null;
+        try {
+            xmlReportStream = new FileInputStream(xmlReport);
```

Comment:
Since the `try` block is here for the stream closing only, the instantiation can be moved in front of the `try` block making the reference non-null simplifying the `finally` block as well.

Reply:
Simpler still in `-source 7`, if we just update to 1.625.x or later:

```java
try (InputStream xmlReportStream = new FileInputStream(xmlReport)) {
    parseSuite(xmlReport, keepLongStdio, r, saxReader.read(xmlReportStream).getRootElement());
}
```